### PR TITLE
rpc: add FeeMsat to ForwardingEvent

### PR DESCRIPTION
**Pull request by sevastos**
In addition to the existing `Fee` field that shows the fees (truncated to satoshis) of a forwarding event, I added a `FeeMsat` field to expose the milli-satoshi amount of those fees.

### CODE REVIEW

File: lnrpc/rpc.proto [hunk 1]
```diff
@@ -1945,6 +1945,9 @@ message ForwardingEvent {
     /// The total fee that this payment circuit carried.
     uint64 fee = 7 [json_name = "fee"];
 
+    /// The total fee (in milli-satoshis) that this payment circuit carried.
```
Comment (halseth) [inline lnrpc/rpc.proto]: Update the comment on the above fields to note they are in satoshis.

File: rpcserver.go [hunk 2]
```diff
@@ -4176,6 +4176,7 @@ func (r *rpcServer) ForwardingHistory(ctx context.Context,
 			AmtIn:     uint64(amtInSat),
 			AmtOut:    uint64(amtOutSat),
 			Fee:       uint64(amtInSat - amtOutSat),
+			FeeMsat:   uint64(event.AmtIn - event.AmtOut),
```
Comment (halseth) [inline rpcserver.go]: Do this calculation above: `feeMsat := event.AmtIn - event.AmtOut`, then set 
```go
			Fee:       uint64(feeMsat.ToSatoshis()),
			FeeMsat:   uint64(feeMsat),
```

**Comment by sevastos** [comment]
@halseth thanks, I pushed the improvements.

File: lnrpc/rpc.proto [hunk 3]
```diff
@@ -1936,15 +1936,18 @@ message ForwardingEvent {
...
-    /// The total amount of the incoming HTLC that created half the circuit.
+    /// The total amount (in satoshis) of the incoming HTLC that created half the circuit.
     uint64 amt_in = 5 [json_name = "amt_in"];
 
-    /// The total amount of the outgoign HTLC that created the second half of the circuit.
+    /// The total amount (in satoshis) of the outgoing HTLC that created the second half of the circuit.
     uint64 amt_out = 6 [json_name = "amt_out"];
 
-    /// The total fee that this payment circuit carried.
+    /// The total fee (in satoshis) that this payment circuit carried.
```
Comment (halseth) [inline lnrpc/rpc.proto]: the `gen_protos.sh` script must be re-run after this change.

Comment (sevastos) [reply lnrpc/rpc.proto]: @halseth done, rebased & ran.

**Comment by sevastos** [comment]
Rebased and rerun `gen_protos.sh`.
However mine reverts the (formatting) changes of db6250803869bff1063ea7a0c45a3b270994aa5d.
Is it possible that the commits on https://github.com/lightningnetwork/lnd/tree/master/lnrpc#generate-protobuf-definitions need an update?

File: lnrpc/rpc.proto [hunk 4]
```diff
@@ -1945,15 +1945,18 @@ message ForwardingEvent {
...
```
Comment (halseth) [inline lnrpc/rpc.proto]: Just noting that we might want to deprecate this fied in the future and move to all fields ending with `_sat` or `_msat`.

**Comment by sevastos** [comment]
Rebased to latest master.

**Comment by sevastos** [comment]
fwiw I've been using this both on testnet and mainnet for a while (to log data) and works fine.

**Comment by halseth** [comment]
Needs a rebase (again) 😅

**Comment by sevastos** [comment]
Rebased :smile:

**Comment by Roasbeef** [comment]
Committed as bd8046ca5c35c735a7f2a2461508a10a050189f0

**Pull request closed without merging**